main.tf

```diff
@@ -2,9 +2,14 @@ data "azurerm_resource_group" "main" {
   name = var.resource_group_name
 }
 
-module "ssh-key" {
-  source         = "./modules/ssh-key"
-  public_ssh_key = var.public_ssh_key == "" ? "" : var.public_ssh_key
+moved {
+  from = module.ssh-key.tls_private_key.ssh
+  to   = tls_private_key.ssh
+}
+
+resource "tls_private_key" "ssh" {
```

Comment:
Yeah that works, but it will also require using index when we reference the resource, and I think an unused private key resource won't do harm, so I choose a shortcut.